debugging/netcoreapp/dockerDebugConfigurationProvider.ts

```diff
@@ -0,0 +1,249 @@
+ *--------------------------------------------------------*/
+
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { CancellationToken, DebugConfiguration, DebugConfigurationProvider, ProviderResult, WorkspaceFolder } from 'vscode';
+import { DebugSessionManager } from './debugSessionManager';
+import { DockerManager, LaunchResult } from './dockerManager';
+import { FileSystemProvider } from './fsProvider';
+import { NetCoreProjectProvider } from './netCoreProjectProvider';
+import { OSProvider, PlatformType  } from './osProvider';
+import { Prerequisite } from './prereqManager';
+
+interface DockerDebugBuildOptions {
+    context?: string;
+    dockerfile?: string;
+    tag?: string;
+    target?: string;
+}
+
+interface DockerDebugRunOptions {
+    containerName?: string;
+    os?: PlatformType;
+}
+
+interface DockerDebugOptions {
+    appFolder?: string;
+    appOutput?: string;
+    appProject?: string;
+    build?: DockerDebugBuildOptions;
+    run?: DockerDebugRunOptions;
+}
+
+interface DebugConfigurationBrowserBaseOptions {
+    enabled?: boolean;
+    command?: string;
+    args?: string;
+}
+
+interface DebugConfigurationBrowserOptions extends DebugConfigurationBrowserBaseOptions {
+    windows?: DebugConfigurationBrowserBaseOptions;
+    osx?: DebugConfigurationBrowserBaseOptions;
+    linux?: DebugConfigurationBrowserBaseOptions;
+}
+
+interface DockerDebugConfiguration extends DebugConfiguration {
+    dockerOptions?: DockerDebugOptions;
+}
+
+export class DockerDebugConfigurationProvider implements DebugConfigurationProvider {
+    constructor(
+        private readonly debugSessionManager: DebugSessionManager,
+        private readonly dockerManager: DockerManager,
+        private readonly fsProvider: FileSystemProvider,
+        private readonly osProvider: OSProvider,
+        private readonly netCoreProjectProvider: NetCoreProjectProvider,
+        private readonly prerequisite: Prerequisite) {
+    }
+
+    public provideDebugConfigurations(folder: WorkspaceFolder | undefined, token?: CancellationToken): ProviderResult<DebugConfiguration[]> {
+        return [
+            {
+                name: 'Docker: Launch .NET Core (Preview)',
+                type: 'docker-netcoreapp',
+                request: 'launch',
+                preLaunchTask: 'build'
+            }
+        ];
+    }
+
+    public resolveDebugConfiguration(folder: WorkspaceFolder | undefined, debugConfiguration: DockerDebugConfiguration, token?: CancellationToken): ProviderResult<DebugConfiguration> {
+        return this.resolveDockerDebugConfiguration(folder, debugConfiguration);
+    }
+
+    private static resolveFolderPath(folderPath: string, folder: WorkspaceFolder): string {
+        return folderPath.replace(/\$\{workspaceFolder\}/g, folder.uri.fsPath);
+    }
+
+    private async resolveDockerDebugConfiguration(folder: WorkspaceFolder, debugConfiguration: DockerDebugConfiguration): Promise<DebugConfiguration> {
```

Comment:
Promise<DebugConfiguration|undefined>.

It would be great if you could temporarily turn on strictNullChecks in tsconfig.json and resolve all errors in your code before merging since we're trying to get the codebase to that point.

Reply:
Made updates to build cleanly with `strictNullChecks` enabled.

Reply:
Thanks!  Much appreciated.